tmx_hexcolor.go

```diff
@@ -0,0 +1,98 @@
+package tiled
+
+import (
+	"encoding/xml"
+	"errors"
+	"fmt"
+	"image/color"
+	"strings"
+)
+
+type HexColor struct {
+	c color.RGBA
+}
+
+func ParseHexColor(s string) (HexColor, error) {
+	c, err := parseHexColor(s)
+	return HexColor{
+		c: c,
+	}, err
+
+}
+
+func NewHexColor(r, g, b, a uint32) HexColor {
+	return HexColor{
+		c: color.RGBA{
+			uint8(r),
+			uint8(g),
+			uint8(b),
+			uint8(a),
+		},
+	}
+}
+
+func (color *HexColor) RGBA() (r, g, b, a uint32) {
+	return color.c.RGBA()
+}
+
+func (color *HexColor) String() string {
+	return fmt.Sprintf("#%x%x%x%x", color.c.R, color.c.G, color.c.B, color.c.A)
```

Comment:
It should probably return just `#%x%x%x` if A is 255

Reply:
I've been thinking about this overnight. If alpha is 255, it may be omitted, but as specified in [tiled docs](https://doc.mapeditor.org/en/stable/reference/tmx-map-format/#map) the color can have an optional alpha.

Turns out that this function was not printing correct strings for values less than 16. I've fixed the issue using the hex package.